pkg/types/baremetal/validation/platform.go

```diff
@@ -338,24 +338,52 @@ func validateNetworkConfig(hosts []*baremetal.Host, fldPath *field.Path) (errors
 	return
 }
 
-// ensure that the bootMode field contains a valid value
-func validateBootMode(hosts []*baremetal.Host, fldPath *field.Path) (errors field.ErrorList) {
+// validateBMC validates all host BMC addresses and calls custom validate function
+func validateBMC(hosts []*baremetal.Host, fldPath *field.Path, validateFn func(host *baremetal.Host, b bmc.AccessDetails, idx int, fldPath *field.Path) field.ErrorList) field.ErrorList {
+	allErrs := field.ErrorList{}
+
 	for idx, host := range hosts {
+		accessDetails, err := bmc.NewAccessDetails(host.BMC.Address, host.BMC.DisableCertificateVerification)
+		if err != nil || accessDetails == nil {
```

Comment:
Wouldn't be better to have a separate check and a specific error message for `accessDetails == nil`?